index.d.ts

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+
+export interface BusyIndicatorProperties {
+  color: string;
+  overlayColor: string;
+  overlayHeight: number;
+  overlayWidth: number;
+  size: 'large' | 'small';
+  startVisible: boolean;
+  text: string;
+  textColor: string;
+  textFontSize: number;
+  textNumberOfLines: number;
+}
+
+export interface BusyIndicatorState {
+  isVisible: boolean;
+}
+
+export class BusyIndicator extends React.Component<BusyIndicatorProperties | {}, BusyIndicatorState> {
```

Comment:
Could you use partial instead of `BusyIndicatorProperties | {}`, feels much nicer IMO.

e.g. `Partial<BusyIndicatorProperties>`

Reply:
Didn't know about `Partial`, what is this doing exactly? Does this allow the `BusyIndicatorProperties` to be `null`?

Reply:
I believe it allows any or all keys to be missing. IE so if a key (prop) is specified, it has to be of that type. If it's missing, then that's fine. I think it ends up roughly the same as adding `?` to all the keys

Reply:
Okay that's cool, thank's for the tip!